x-pack/plugins/apm/public/components/app/error_group_overview/index.tsx

```diff
@@ -99,47 +94,43 @@ export function ErrorGroupOverview() {
                 },
               },
             }
-          ).then((response) => {
-            const currentPageGroupIds = orderBy(
-              response.errorGroups,
-              sortField,
-              sortDirection
-            )
-              .slice(page * pageSize, (page + 1) * pageSize)
-              .map(({ groupId }) => groupId)
-              .sort();
-
-            return {
-              // Everytime the main statistics is refetched, updates the requestId making the comparison API to be refetched.
-              requestId: uuid(),
-              errorGroupMainStatistics: response.errorGroups,
-              currentPageGroupIds,
-            };
-          });
+          );
         }
       },
-      [
-        environment,
-        kuery,
-        serviceName,
-        start,
-        end,
-        sortField,
-        sortDirection,
-        page,
-        pageSize,
-      ]
+      [environment, kuery, serviceName, start, end, sortField, sortDirection]
     );
 
-  const { requestId, errorGroupMainStatistics, currentPageGroupIds } =
-    errorGroupListData;
+  const { errorGroups } = errorGroupListData;
+
+  const { tableItems, tablePagination, tableSort, onTableChange, requestId } =
+    useTableSortAndPagination(
+      {
+        items: errorGroups,
+        pagination: {
+          pageIndex: page,
+          pageSize: pageSize || PAGE_SIZE,
+          showPerPageOptions: true,
+        },
+        sorting: {
+          sort: {
+            field:
+              (sortField as
+                | keyof ErrorGroupMainStatistics['errorGroups'][0]
+                | undefined) || SORT_FIELD,
```

Comment:
Instead of forcing the type system, it would be great if we can change the original type of `sortField` here:

https://github.com/elastic/kibana/blob/10ef628e12af1556ef6d68d03d5c076b5500da89/x-pack/plugins/apm/server/routes/errors/route.ts#L29

Reply:
You'll probably need to define the following as io-ts, then infer the ts type from that:

https://github.com/elastic/kibana/blob/10ef628e12af1556ef6d68d03d5c076b5500da89/x-pack/plugins/apm/server/routes/errors/route.ts#L41-L49

Reply:
Agreed that it would be better, but don't you think it's a bit out of scope of this PR? I want to have it as clean as possible so people focus on the new hook, not on other changes I did. I'm happy to change it on a follow-up PR though.

Reply:
Follow-up is fine